homeassistant/helpers/entity_registry.py

```diff
@@ -1282,6 +1305,14 @@ def async_entries_for_area(
     return registry.entities.get_entries_for_area_id(area_id)
 
 
+@callback
+def async_entries_for_label(
+    registry: EntityRegistry, label_id: str
+) -> list[RegistryEntry]:
+    """Return entries that match a label."""
+    return [entry for entry in registry.entities.values() if label_id in entry.labels]
```

Comment:
This one might get expensive if we call it a lot since it has to do the linear search.  We can always add an index if that turns out to be the case